subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1920,6 +1927,7 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 		  "[-w, --ieee-80211w]: MFP (optional: needs security type to be specified)\n"
 		  ": 0:Disable, 1:Optional, 2:Required.\n"
 		  "[-m, --bssid]: MAC address of the AP (BSSID).\n"
+		  "[-t, --timeout]: Timeout for the connection attempt.\n"
```

Comment:
```suggestion
		  "[-t, --timeout]: Timeout for the connection attempt (in seconds).\n"
```

Reply:
Any range for this timeout? And do we need a magic value to indicate no timeout? e.g., `-1`?

Reply:
Value of 0 (default for params->timeout) indicates no timeout.

Reply:
About the range for timeout, since `timeout=0` means connection attempt may go forever, don't see a valid upper limit that can be defined for `-t` option.

Reply:
Forever in Zephyr is `-1` (`SYS_FOREVER_MS`) that is the reason why `timeout` is `int`.